pkg/helpers/autoimport.go

```diff
@@ -6,58 +6,248 @@ package helpers
+	importSecret, err := i.informerHolder.ImportSecretLister.Secrets(clusterName).Get(importSecretName)
+	if errors.IsNotFound(err) {
+		return reconcile.Result{},
+			NewManagedClusterImportSucceededCondition(
+				metav1.ConditionFalse,
+				constants.ConditionReasonManagedClusterImporting,
+				"Wait for import secret",
+			), false, currentRetry, nil
+	}
+	if err != nil {
+		return reconcile.Result{},
+			NewManagedClusterImportSucceededCondition(
+				metav1.ConditionFalse,
+				constants.ConditionReasonManagedClusterImporting,
+				fmt.Sprintf("Get import secret failed: %v. Will retry", err),
+			), false, currentRetry, err
+	}
+
+	currentRetry++
+	modified, err := i.applyResourcesFunc(backupRestore, clientHolder, restMapper, i.recorder, importSecret)
+	if err != nil {
+		condition := NewManagedClusterImportSucceededCondition(
+			metav1.ConditionFalse,
+			constants.ConditionReasonManagedClusterImportFailed,
+			fmt.Sprintf("Try to import managed cluster, retry times: %d/%d, error: %v",
+				currentRetry, totalRetry, err),
+		)
+
+		if ContainAuthError(err) {
+			// return message reflects the auto import secret is invalid, so the user knows that
+			// a correct secret needs to be re-provided
+			condition.Message = fmt.Sprintf(
+				"AutoImportSecretInvalid %s/%s; please check its permission, apply resources error: %v",
+				managedClusterKubeClientSecret.Namespace, managedClusterKubeClientSecret.Name, err)
+			return reconcile.Result{}, condition, modified, currentRetry, nil
+		}
+
+		if ContainInternalServerError(err) {
+			// might be some internal server error, does not take up retry times
+			condition.Reason = constants.ConditionReasonManagedClusterImporting
+			condition.Message = fmt.Sprintf(
+				"Try to import managed cluster, apply resources error: %v. Will Retry", err)
+			return reconcile.Result{}, condition, modified, lastRetry, err
+		}
+
+		if currentRetry < totalRetry {
+			// maybe some network issue, retry after 10 second
+			condition.Reason = constants.ConditionReasonManagedClusterImporting
+			return reconcile.Result{RequeueAfter: 10 * time.Second},
+				condition, modified, currentRetry, nil
+		}
+
+		return reconcile.Result{},
+			condition, modified, currentRetry, nil
+	}
+
+	return reconcile.Result{},
+		NewManagedClusterImportSucceededCondition(
+			metav1.ConditionFalse,
+			constants.ConditionReasonManagedClusterImporting,
+			conditionMessageImportingResourcesApplied,
+		), modified, currentRetry, nil
+}
+
+const (
+	conditionMessageImportingResourcesApplied = "Importing resources are applied, wait for resources be available"
+)
+
+func ImportingResourcesApplied(condition *metav1.Condition) bool {
+	if condition != nil && condition.Type == constants.ConditionManagedClusterImportSucceeded &&
+		condition.Reason == constants.ConditionReasonManagedClusterImporting &&
+		condition.Message == conditionMessageImportingResourcesApplied {
```

Comment:
do we need judge message?

Reply:
yes, because there are other cases the reason is `importing`,  so I think we can add a specific method to set the the ImportSucceeded condition to ResourceApplied